Move parsedSearchQuery data to query state store

fkling opened the pull request:
This PR moves `parsedSearchQuery`, which is currently created and passed down from `Layout.tsx` to the query state store. I'm also renaming it to `searchQueryFromURL` to better express what this value represents (the current name is easy to confuse with parsing the query to a token representation).

This used to be stored in `SourcegraphWebApp`s state, but @valerybugakov removed this already in #29176. However, I still think that threading it through multiple layers of components is not necessary.

It looks like there are only a few number of places where access to the URL search query is needed:
- `GlobalNavbar` updates the "internal" query state when the query from the URL changes (this will likely be refactored once search context is also moved to the query store)
- `routes.tsx` decides whether to to show the homepage or the result page based on whether a query is set or not (but maybe there is a better way?)
- `DidYouMean` should operate on the submitted query, not the current query from the search input

Removed from:
- `SearchPageInput` only used this value to decide whether or not to show the onboarding tour, but this is better determined by the parent component (`SearchPage`)

fkling commented:
Mmh, somehow this seems to cause the one puppeteer test to fail that checks whether suggestions are shown in the search input... will have to investigate.

fkling commented on client/web/src/integration/search.test.ts:
```diff
@@ -198,7 +198,7 @@ describe('Search', () => {
             // File autocomplete from repo search bar
             await driver.page.waitForSelector('#monaco-query-input')
-            await driver.page.focus('#monaco-query-input')
+            await driver.page.click('#monaco-query-input')
```
This seems to fix the test, at least locally. This is basically what `driver.replaceText` uses as well. However, I don't know why/how this would be affected by this PR.

limitedmage commented on client/web/src/search/home/SearchPageInput.tsx:
```diff
@@ -77,19 +77,11 @@ export const SearchPageInput: React.FunctionComponent<Props> = (props: Props) =>
     const quickLinks =
         (isSettingsValid<Settings>(props.settingsCascade) && props.settingsCascade.final.quicklinks) || []
 
-    // This component is also used on the CommunitySearchContextPage.
-    // The search onboarding tour should only be shown on the homepage.
-    const isHomepage = useMemo(() => props.location.pathname === '/search' && !props.parsedSearchQuery, [
-        props.location.pathname,
-        props.parsedSearchQuery,
-    ])
-    const showOnboardingTour = useExperimentalFeatures(features => features.showOnboardingTour ?? false) && isHomepage
```
Nice removal, passing it as a prop is a much nicer solution 👍

limitedmage commented on client/web/src/search/results/StreamingSearchResults.story.tsx:
```diff
@@ -95,27 +94,15 @@ add('no results', () => {
     return <WebStory>{() => <StreamingSearchResults {...defaultProps} streamSearch={() => of(result)} />}</WebStory>
 })
 
-add('diffs tab selected, user logged in', () => (
-    <WebStory>
-        {() => <StreamingSearchResults {...defaultProps} parsedSearchQuery="r:golang/oauth2 test f:travis type:diff" />}
-    </WebStory>
-))
-
-add('code tab selected, user logged in', () => (
-    <WebStory>
-        {() => <StreamingSearchResults {...defaultProps} parsedSearchQuery="r:golang/oauth2 test f:travis" />}
-    </WebStory>
-))
```
Can one (or both) of these tests be added back in just to have a story with "normal" search results? I'm guessing the tabs don't matter anymore since they were removed in favor of the sidebar, so I think just the "code" one should be OK.

fkling replied:
I think that's what 

```
add('standard render', () => <WebStory>{() => <StreamingSearchResults {...defaultProps} />}</WebStory>)
```

already does. At least it looked the same as these two.

fkling commented:
This PR moves parsedSearchQuery, which is currently created and passed down from Layout.tsx to the query state store. I'm also renaming it to searchQueryFromURL to better express what this value represents (the current name is easy to confuse with parsing the query to a token representation).

This used to be stored in SourcegraphWebApps state, but @valerybugakov removed this already in #29176. However, I still think that threading it through multiple layers of components is not necessary.

It looks like there are only a few number of places where access to the URL search query is needed:

- GlobalNavbar updates the "internal" query state when the query from the URL changes (this will likely be refactored once search context is also moved to the query store)
- routes.tsx decides whether to to show the homepage or the result page based on whether a query is set or not (but maybe there is a better way?)
- DidYouMean should operate on the submitted query, not the current query from the search input

Removed from:

- SearchPageInput only used this value to decide whether or not to show the onboarding tour, but this is better determined by the parent component (SearchPage)
